stdlib/http/src/main/ballerina/src/http/cookie/cookieStore.bal

```diff
@@ -114,39 +140,150 @@ public type CookieStore object {
     #
     # + return - Array of all the cookie objects
     public function getAllCookies() returns Cookie[] {
-        return self.allSessionCookies;
+        var persistentCookieHandler = self.persistentCookieHandler;
+        Cookie[] allCookies = [];
+        foreach var cookie in self.allSessionCookies {
+            allCookies.push(cookie);
+        }
+        if (persistentCookieHandler is PersistentCookieHandler) {
+            var result = persistentCookieHandler.getAllCookies();
+            if (result is error) {
+                log:printError("Error in getting persistent cookies: ", result);
```

Comment:
we call this function when a client sends a request , has to get cookies from the cookie store. So wants to return at least session cookies in a case we get an error from persistent cookies.if we,return an error because of the persistent cookies, session cookies also will not be returned. That's why, used a log to make aware about the error in persistent cookies and return only the session cookies.

Reply:
Right.